app/models/workers_scaling/algorithms/sample_algorithm.rb

```diff
@@ -160,9 +157,19 @@ def decrease_computational_power
     # If average_throughput is 0, throughput_needed value is ignored and exactly one Worker is added
     # Returns predicted throughput growth
     def add_workers(infrastructure, average_throughput = 0, throughput_needed = 0)
-      workers_needed = average_throughput > 0 ? (throughput_needed / average_throughput).ceil : 1
-      LOGGER.debug "Starting #{workers_needed} Workers on infrastructure: #{infrastructure}"
-      @resources_interface.schedule_workers(workers_needed, infrastructure).count * average_throughput
+      workers_needed = if average_throughput > 0
+                         if throughput_needed == Float::INFINITY
+                           Float::INFINITY
+                         else
+                           (throughput_needed / average_throughput).ceil
+                         end
+                       else
+                         1
+                       end
+      LOGGER.debug "Trying to schedule #{workers_needed} Workers on infrastructure: #{infrastructure}"
+      scheduled = @resources_interface.schedule_workers(workers_needed, infrastructure).count
+      LOGGER.debug "Total of #{scheduled} Workers starting on infrastructure: #{infrastructure}"
```

Comment:
Tutaj te logi są celowo, bo widać ile algorytm chce, a ile dostał. To poziom debug, teoretycznie nie przeszkadza wcale, ale trzeba będzie potem przerobić bo wszystko w WSS jest na debug.